### PR TITLE
Default async/await implementation

### PR DESCRIPTION
**Pull request by laqiguy**
There is no need to do it yourself. Default implementation uses already implemented methods with closures and wrap them into async/await with cancelation.

### CODE REVIEW

File: Sources/Apexy/AsyncAwaitHelper.swift [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+//
+
+import Foundation
+
+@available(macOS 10.15, iOS 13, watchOS 6, tvOS 13, *)
+public enum AsyncAwaitHelper {
+    public typealias ContentContinuation<T> = CheckedContinuation<T, Error>
+
+    public static func adaptToAsync<T>(dataTaskClosure: (ContentContinuation<T>) -> Progress) async throws -> T {
+            let progressWrapper = ProgressWrapper()
```
Comment (vani2) [inline Sources/Apexy/AsyncAwaitHelper.swift]: Indent is too big

File: Sources/Apexy/Client.swift [hunk 2]
```diff
@@ -37,3 +37,33 @@ public protocol Client: AnyObject {
...
+    
+    func upload<T>(_ endpoint: T) async throws -> T.Content where T: UploadEndpoint {
+        return try await AsyncAwaitHelper.adaptToAsync(dataTaskClosure: { continuation in
+            return upload(endpoint) { result in
+                switch result {
+                case .success(let content):
+                    return continuation.resume(returning: content)
+                case .failure(let error):
+                    return continuation.resume(throwing: error)
+                }
```
Comment (vani2) [inline Sources/Apexy/Client.swift]: Can it be done with `continuation.resume(with: result)`?

Comment (vani2) [reply Sources/Apexy/Client.swift]: Also, looks like we can omit `return`

File: Sources/ApexyURLSession/URLSession+AsyncAwait13.swift [hunk 3]
```diff
@@ -13,65 +13,54 @@ import Foundation
...
-            })
-    }
     
     public func data(
         for request: URLRequest,
         delegate: URLSessionTaskDelegate? = nil) async throws -> (Data, URLResponse) {
-            return try await adaptToAsync(dataTaskClosure: { continuation in
-                return self.dataTask(with: request) { data, response, error in
+            return try await AsyncAwaitHelper.adaptToAsync(dataTaskClosure: { continuation in
+                let task = self.dataTask(with: request) { data, response, error in
```
Comment (subdan) [inline Sources/ApexyURLSession/URLSession+AsyncAwait13.swift]: No memory leaks here?

File: Sources/Apexy/AsyncAwaitHelper.swift (hunk 1 above)
Comment (laqiguy) [reply Sources/Apexy/AsyncAwaitHelper.swift]: fixed

File: Sources/Apexy/Client.swift (hunk 2 above)
Comment (laqiguy) [reply Sources/Apexy/Client.swift]: fixed. it looks much better now)

File: Sources/ApexyURLSession/URLSession+AsyncAwait13.swift (hunk 3 above)
Comment (laqiguy) [reply Sources/ApexyURLSession/URLSession+AsyncAwait13.swift]: i removed `self`. i think there is no leak now

**Review by subdan — APPROVED** [review]

**Comment by laqiguy** [comment]
@vani2 Please, look into changes your comments caused.

**Review by vani2 — APPROVED** [review]

**Pull request merged**